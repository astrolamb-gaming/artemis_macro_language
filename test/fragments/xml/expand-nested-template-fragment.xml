<mission_data>
  <templates>
    <template name="tonnage:destroy">
      <params>
        <param name="name" description="The name of the ship"/>
        <param name="weight" description="The weight in tonnage of the ship"/>
        <param name="race" description="The race text for the ship"/>
        <param name="hull" description="The hull text for the ship" default=""/>
      </params>
      <event name="${name} Destroyed">
        <if_variable name="${name}_surrendered" comparator="EQUALS" value="0"/>
        <if_variable name="${name}_dead" comparator="EQUALS" value="0"/>
        <if_not_exists name="${name}"/>
        <set_variable name="${name}_dead" value="1"/>
        <set_variable name="report_score" value="1"/>
        <set_timer name="report_timer" seconds="8"/>
        <set_variable name="Tonnage" value="Tonnage+${weight}"/>
        <big_message title="${race} ${hull} ${name} destroyed" subtitle1="${weight} kilotons" subtitle2=""/>
        <set_variable name="report_score" value="1"/>
        <set_timer name="report_timer" seconds="8"/>
        <log text="${race} ${hull} ${name} destroyed"/>
      </event>
    </template>
    <template name="tonnage:surrender">
      <params>
        <param name="name" description="The name of the ship"/>
        <param name="weight" description="The weight in tonnage of the ship"/>
        <param name="surrender" description="The surrender bonus of the ship"/>
        <param name="race" description="The race text for the ship"/>
        <param name="hull" description="The hull text for the ship"  default=""/>
        <param name="fleet" description="The fleet ID (For bonus fleets)" default=""/>
      </params>
      <event name="${name} Surrendered">
        <if_variable name="${name}_surrendered" comparator="EQUALS" value="0"/>
        <if_object_property property="hasSurrendered" name="${name}" comparator="EQUALS" value="1"/>
        <set_variable name="${name}_surrendered" value="1"/>
        <set_object_property property="topSpeed" value="0.2" name="${name}"/>
        <set_variable name="Tonnage" value="Tonnage+${weight}+${surrender}"/>
        <big_message title="${race} ${hull} surrendered" subtitle1="${weight} kilotons" subtitle2="+${surrender} kt surrender bonus"/>
        <set_variable name="report_score" value="1"/>
        <set_timer name="report_timer" seconds="8"/>
        <log text="${race} ${hull} ${name} Surrendered"/>
      </event>
    </template>
    <template name="tonnage:points">
      <expand _template="tonnage:surrender" />
      <expand _template="tonnage:destroy" />
    </template>
  </templates>

    <values>
        <range name="hunters">
            <value name="Hunter[1]" timer="70" x="58300" y="57" z="78300" weight="40" surrender="20" />
            <value name="Hunter[2]" timer="80" x="59300" y="57" z="23500" weight="40" surrender="20" />
            <value name="Hunter[3]" timer="90" x="38200" y="-57" z="43500" weight="40" surrender="20" />
            <value name="Hunter[4]" timer="100" x="58200" y="-245" z="43500" weight="40" surrender="20" />
        </range>
    </values>
    <repeat _range="hunters">
        <!--Override text to match original  -->
        <tonnage:points race="Enemy" hull="" />
    </repeat>
</mission_data>