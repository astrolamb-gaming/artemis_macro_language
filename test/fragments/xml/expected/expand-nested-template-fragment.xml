<mission_data>
  <event name="Hunter[1] Surrendered">
    <if_variable name="Hunter[1]_surrendered" comparator="EQUALS" value="0"/>
    <if_object_property property="hasSurrendered" name="Hunter[1]" comparator="EQUALS" value="1"/>
    <set_variable name="Hunter[1]_surrendered" value="1"/>
    <set_object_property property="topSpeed" value="0.2" name="Hunter[1]"/>
    <set_variable name="Tonnage" value="Tonnage+40+20"/>
    <big_message title="Enemy  surrendered" subtitle1="40 kilotons" subtitle2="+20 kt surrender bonus"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Enemy  Hunter[1] Surrendered"/>
  </event>
  <event name="Hunter[1] Destroyed">
    <if_variable name="Hunter[1]_surrendered" comparator="EQUALS" value="0"/>
    <if_variable name="Hunter[1]_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="Hunter[1]"/>
    <set_variable name="Hunter[1]_dead" value="1"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <set_variable name="Tonnage" value="Tonnage+40"/>
    <big_message title="Enemy  Hunter[1] destroyed" subtitle1="40 kilotons"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Enemy  Hunter[1] destroyed"/>
  </event>
  <event name="Hunter[2] Surrendered">
    <if_variable name="Hunter[2]_surrendered" comparator="EQUALS" value="0"/>
    <if_object_property property="hasSurrendered" name="Hunter[2]" comparator="EQUALS" value="1"/>
    <set_variable name="Hunter[2]_surrendered" value="1"/>
    <set_object_property property="topSpeed" value="0.2" name="Hunter[2]"/>
    <set_variable name="Tonnage" value="Tonnage+40+20"/>
    <big_message title="Enemy  surrendered" subtitle1="40 kilotons" subtitle2="+20 kt surrender bonus"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Enemy  Hunter[2] Surrendered"/>
  </event>
  <event name="Hunter[2] Destroyed">
    <if_variable name="Hunter[2]_surrendered" comparator="EQUALS" value="0"/>
    <if_variable name="Hunter[2]_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="Hunter[2]"/>
    <set_variable name="Hunter[2]_dead" value="1"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <set_variable name="Tonnage" value="Tonnage+40"/>
    <big_message title="Enemy  Hunter[2] destroyed" subtitle1="40 kilotons"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Enemy  Hunter[2] destroyed"/>
  </event>
  <event name="Hunter[3] Surrendered">
    <if_variable name="Hunter[3]_surrendered" comparator="EQUALS" value="0"/>
    <if_object_property property="hasSurrendered" name="Hunter[3]" comparator="EQUALS" value="1"/>
    <set_variable name="Hunter[3]_surrendered" value="1"/>
    <set_object_property property="topSpeed" value="0.2" name="Hunter[3]"/>
    <set_variable name="Tonnage" value="Tonnage+40+20"/>
    <big_message title="Enemy  surrendered" subtitle1="40 kilotons" subtitle2="+20 kt surrender bonus"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Enemy  Hunter[3] Surrendered"/>
  </event>
  <event name="Hunter[3] Destroyed">
    <if_variable name="Hunter[3]_surrendered" comparator="EQUALS" value="0"/>
    <if_variable name="Hunter[3]_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="Hunter[3]"/>
    <set_variable name="Hunter[3]_dead" value="1"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <set_variable name="Tonnage" value="Tonnage+40"/>
    <big_message title="Enemy  Hunter[3] destroyed" subtitle1="40 kilotons"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Enemy  Hunter[3] destroyed"/>
  </event>
  <event name="Hunter[4] Surrendered">
    <if_variable name="Hunter[4]_surrendered" comparator="EQUALS" value="0"/>
    <if_object_property property="hasSurrendered" name="Hunter[4]" comparator="EQUALS" value="1"/>
    <set_variable name="Hunter[4]_surrendered" value="1"/>
    <set_object_property property="topSpeed" value="0.2" name="Hunter[4]"/>
    <set_variable name="Tonnage" value="Tonnage+40+20"/>
    <big_message title="Enemy  surrendered" subtitle1="40 kilotons" subtitle2="+20 kt surrender bonus"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Enemy  Hunter[4] Surrendered"/>
  </event>
  <event name="Hunter[4] Destroyed">
    <if_variable name="Hunter[4]_surrendered" comparator="EQUALS" value="0"/>
    <if_variable name="Hunter[4]_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="Hunter[4]"/>
    <set_variable name="Hunter[4]_dead" value="1"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <set_variable name="Tonnage" value="Tonnage+40"/>
    <big_message title="Enemy  Hunter[4] destroyed" subtitle1="40 kilotons"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Enemy  Hunter[4] destroyed"/>
  </event>
</mission_data>